Python/Tests/DebuggerTests/MiniDumpWriter.cs

```diff
@@ -0,0 +1,92 @@
+// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using TestUtilities;
+
+namespace DebuggerTests {
+    /// <summary>
+    /// Writes a crash dump of the designated process on dispose, unless canceled.
+    /// </summary>
+    internal sealed class MiniDumpWriter : IDisposable {
+        // Full memory dumps are 3 orders of magnitude larger than stack dumps, so
+        // they're not enabled by default, but can be switched on for debugging.
+        private const MINIDUMP_TYPE minidumpType =
+            MINIDUMP_TYPE.MiniDumpNormal;
+            //MINIDUMP_TYPE.MiniDumpWithFullMemory;
+
+        private Process _process;
+
+        public MiniDumpWriter(Process process) {
+            _process = process;
+        }
+
+        public void Cancel() {
+            _process = null;
+        }
+
+        public void Dispose() {
+            if (_process == null || _process.HasExited) {
+                return;
+            }
+
+            IntPtr handle;
+            uint id;
+            try {
+                handle = _process.Handle;
+                id = (uint)_process.Id;
+            } catch (InvalidOperationException) {
+                return;
+            }
+
+            var dumpDir = TestData.GetTempPath(randomSubPath: true);
+            var dumpPath = Path.Combine(dumpDir, "dump.dmp");
```

Comment:
I don't know how frequently you had to review dump files, but would it be useful to have the ability to control the dump file name? Constructor could accept a name, and tests could pass TestContext.FullyQualifiedTestClassName.